src/main/java/com/epam/esm/config/security/SecurityConfig.java

```diff
@@ -0,0 +1,65 @@
+package com.epam.esm.config.security;
+
+import com.epam.esm.domain.Role;
+import com.epam.esm.security.filter.JwtTokenFilter;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+@RequiredArgsConstructor
+public class SecurityConfig {
+    private final JwtTokenFilter jwtTokenFilter;
+
+    private final AuthenticationProvider authenticationProvider;
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http = http.cors()
```

Comment:
Why you do these strange assigns again and again? there is no need of assigning as far as you know. Example: .csrf()
                .disable()
                .authorizeHttpRequests()
                .requestMatchers("/api/v2/auth/**").permitAll()
                .requestMatchers(HttpMethod.GET, "/api/v2/certificates/**").permitAll()
                .requestMatchers(HttpMethod.GET,"/api/v2/orders", "/api/v2/tags", "/api/v2/users").hasAnyAuthority("USER", "ADMIN")
                .requestMatchers(HttpMethod.POST,"/api/v2/orders").hasAnyAuthority("USER", "ADMIN")
                .requestMatchers("/api/v2/**").hasAuthority("ADMIN")
                .anyRequest()
                .fullyAuthenticated()
                .and()
                .sessionManagement()
                .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
                .and()
                .authenticationProvider(authenticationProvider)
                .addFilterBefore(jwtAuthenticationFilter, UsernamePasswordAuthenticationFilter.class)
                .exceptionHandling()
                .authenticationEntryPoint(authEntryPoint)
                .and()
                .build();

Reply:
Probably because I wanted to separate code onto few logic blocks. Thought that it would be easier to read and maintain. Already fixed that in accordance with your example.

Reply:
Actually it's already easy to read and you don't need to separate this